arrow/src/array/array_union.rs

```diff
@@ -83,6 +27,15 @@ use std::any::Any;
 use std::mem::size_of;
 
 /// An Array that can represent slots of varying types.
+///
+/// Each slot in a `UnionArray` can have a value chosen from a number of types.  Each of the
+/// possible types are named like the fields of a [`StructArray`](crate::array::StructArray).
+/// A `UnionArray` can have two possible memory layouts, "dense" or "sparse".  For more information
+/// on please see the [specification](https://arrow.apache.org/docs/format/Columnar.html#union-layout).
+///
+/// Builders are provided for `UnionArray`'s involving primitive types.  `UnionArray`'s of nested
```

Comment:
```suggestion
/// [`UnionBuilder`]can be used to create  `UnionArray`'s of primitive types.  `UnionArray`'s of nested
```

it might be nice to add a link to the UnionBuilder here so when people are browsing the docs they can click and go to the right place